packages/babel-generator/src/generators/expressions.js

```diff
@@ -139,6 +139,43 @@ export function OptionalCallExpression(node: Object) {
   this.token(")");
 }
 
+export function EventualMemberExpression(node: Object) {
+  this.print(node.object, node);
+
+  if (!node.computed && t.isMemberExpression(node.property)) {
+    throw new TypeError("Got a MemberExpression for MemberExpression property");
+  }
+
+  let computed = node.computed;
+  if (t.isLiteral(node.property) && typeof node.property.value === "number") {
+    computed = true;
+  }
+  this.token("~.");
+
+  if (computed) {
+    this.token("[");
+    this.print(node.property, node);
+    this.token("]");
+  } else {
+    if (!node.optional) {
```

Comment:
The `optional` property has been removed from AST so `!node.optional` is always `true`.